openHarmony/classes/oNode.js

```diff
@@ -108,8 +121,200 @@ function oNode ( path, oSceneObject ){
+    try{
+      // Create temp group/node to discover attributes for this type
+      // Use unique name with timestamp to avoid collisions
+      var scanGroupName = '_OH_TYPE_SCAN_' + (new Date()).getTime();
+      tempGroupPath = node.add('Top', scanGroupName, 'GROUP', 0, 0, 0);
+      if (!tempGroupPath) {
+        // Failed to create temp group, fall back to base class
+        return oNode;
+      }
+      tempNodePath = node.add(tempGroupPath, 'temp', type, 0, 0, 0);
+      if (!tempNodePath) {
+        // Failed to create temp node, clean up and fall back
+        try { node.deleteNode(tempGroupPath); } catch(e) {}
+        return oNode;
+      }
+
+      var attrList = node.getAttrList(tempNodePath, 1);
+      // If attrList is null/undefined/empty, we'll use fallback keywords below
+      var baseKeywords = {};
+      var attrLength = (attrList && attrList.length) ? attrList.length : 0;
+      for (var j = 0; j < attrLength; j++) {
+        try{
+          var attr = attrList[j];
+          if (!attr || typeof attr.keyword !== 'function') continue;
+          var kw = attr.keyword().toLowerCase();
+          // Only keep top-level part (before dot) to define shorthand entry point
+          var base = kw.split('.')[0];
+          if (base === '3dpath') base = 'path3d';
+          if (base) baseKeywords[base] = true;
+        }catch(e){
+          // Skip attributes that fail
+        }
+      }
+      
+      // If no keywords found from scan, add common ones as fallback
+      var keywordCount = 0;
+      for (var k in baseKeywords) { keywordCount++; }
+      if (keywordCount === 0) {
+        // Common node attributes that most node types have
+        var commonKeywords = ['position', 'scale', 'rotation', 'offset', 'skew', 'pivot'];
+        for (var c = 0; c < commonKeywords.length; c++) {
+          baseKeywords[commonKeywords[c]] = true;
+        }
+      }
+
+      // Build per-type subclass
+      var TypeCtor = function(path, oSceneObject){
+        return oNode._init.call(this, path, oSceneObject);
+      };
+      TypeCtor.prototype = Object.create(oNode.prototype);
+      TypeCtor.prototype.constructor = TypeCtor;
+
+      // Define prototype shorthand getters that trigger lazy load
+      for (var kw in baseKeywords) {
+        if (TypeCtor.prototype.hasOwnProperty(kw)) continue;
+        (function(kw){
+          Object.defineProperty(TypeCtor.prototype, kw, {
+            configurable: true,
+            enumerable: false,
+            get: function(){
+              // Trigger lazy loading; attributes getter will install instance getters
+              var attrs = this.attributes;
+              if (!attrs) return undefined;
+              
+              // After lazy loading, instance getter should exist (created by setAttrGetterSetter)
+              // Check and call it directly to get proper value with sub-attribute handling
+              var desc = Object.getOwnPropertyDescriptor(this, kw);
```

Comment:
should it not be 
```js
var desc = Object.getOwnPropertyDescriptor(attrs, kw);
```
here?

I'm a little confused about why we're not just accessing the attribute from the attribute object using the kw.